libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/TemplateEngineLanguageGenerator.cs

```diff
@@ -11,44 +13,29 @@ public class TemplateEngineLanguageGenerator : ILanguageGenerator
     {
         private const string DEFAULTLABEL = "Unknown";
         private TemplateEngine engine;
+        private readonly ResourceExplorer resourceExplorer;
```

Comment:
We should not leak the abstraction of ResourceExplorer and Resource into this layer.  Can we just pass what this Generator need? content + id + resolver, we can change the resolve interface, but not pass the whole resource explorer in